contracts/Rollup.sol

```diff
@@ -174,10 +182,15 @@ contract Rollup is Ownable, RollupHelpers, RollupInterface {
         Entry memory onChainHashData = buildOnChainData(fillingOnChainTxsHash, txData, loadAmount,
             ethAddress, babyPubKey[0], babyPubKey[1]);
         fillingOnChainTxsHash = hashEntry(onChainHashData);
-        // Update total on-chain fees
-        totalFillingOnChainFee += onChainFee;
         // Update number of on-chain transactions
         currentOnChainTx++;
+
+        //Burn some fee depending on the currentOnChainTx, Max value onChainFee / 3
+        uint256 burnedFee = onChainFee * currentOnChainTx / (MAX_ONCHAIN_TX * 3);
```

Comment:
I would add brackets to make operations clear:
`uint256 burnedFee = (onChainFee * currentOnChainTx) / (MAX_ONCHAIN_TX * 3)`